faster_sam/middleware.py

```diff
@@ -5,10 +5,49 @@
 
 
 class RemovePathMiddleware(BaseHTTPMiddleware):
+    """
+    Middleware for removing a specified part of the request path.
+
+    e.g
+
+    >>> app = FastAPI()
+    >>> app.add_middleware(RemovePathMiddleware, path="/foo")  # Removes /foo from requests path
+    >>> # /foo/bar ---> midleware ----> /bar
+    >>> @app.get("/bar")
+    ... def bar():
+    ...     return {"message": "Responding to GET /foo/bar}
+    """
+
     def __init__(self, app: ASGIApp, path: str) -> None:
+        """
+        Initializes the RemovePathMiddleware.
+
+        Parameters
+        ----------
+        app : ASGIApp
+            The application that the middleware is applied.
+        path : str
+            The part of the path to be removed from incoming requests.
+        """
         super().__init__(app, self.dispatch)
         self.path = path
 
     async def dispatch(self, request: Request, call_next: RequestResponseEndpoint) -> Response:
+        """
+        Method that removes the specified part of the request path.
```

Comment:
Novamente, acho que linguagem imperativa funciona melhor para este tipo de documentação.

_Removes the specified part of the request path._